application/src/main/java/com/ericsson/bss/cassandra/ecchronos/application/spring/BeanConfigurator.java

```diff
@@ -0,0 +1,281 @@
+    ) throws IOException
+    {
+        return getDistributedJmxConnection(
+                jmxSecurity::get, distributedNativeConnectionProvider, eccNodesSync);
+    }
+
+    private Security getSecurityConfig() throws ConfigurationException
+    {
+        return ConfigurationHelper.DEFAULT_INSTANCE.getConfiguration(SECURITY_FILE, Security.class);
+    }
+
+    private Config getConfiguration() throws ConfigurationException
+    {
+        return ConfigurationHelper.DEFAULT_INSTANCE.getConfiguration(CONFIGURATION_FILE, Config.class);
+    }
+
+    private DistributedNativeConnectionProvider getDistributedNativeConnection(
+            final Config config,
+            final Supplier<Security.CqlSecurity> securitySupplier
+    )
+    {
+        Supplier<CqlTLSConfig> tlsSupplier = () -> securitySupplier.get().getCqlTlsConfig();
+        CertificateHandler certificateHandler = createCertificateHandler(tlsSupplier);
+        return new AgentNativeConnectionProvider(config, securitySupplier, certificateHandler);
+    }
+
+    private DistributedJmxConnectionProvider getDistributedJmxConnection(
+            final Supplier<Security.JmxSecurity> securitySupplier,
+            final DistributedNativeConnectionProvider distributedNativeConnectionProvider,
+            final EccNodesSync eccNodesSync
+    ) throws IOException
+    {
+        return new AgentJmxConnectionProvider(
+                securitySupplier, distributedNativeConnectionProvider, eccNodesSync);
+    }
+
+    private void refreshSecurityConfig(
+            final Consumer<Security.CqlSecurity> cqlSetter,
+            final Consumer<Security.JmxSecurity> jmxSetter
+    )
+    {
+        try
+        {
+            Security security = getSecurityConfig();
+            cqlSetter.accept(security.getCqlSecurity());
+            jmxSetter.accept(security.getJmxSecurity());
+        }
+        catch (ConfigurationException e)
+        {
+            LOG.warn("Unable to refresh security config");
+        }
+    }
+
+    private static CertificateHandler createCertificateHandler(
+            final Supplier<CqlTLSConfig> tlsSupplier
+    )
+    {
+        return new ReloadingCertificateHandler(tlsSupplier);
+    }
+
+    private EccNodesSync getEccNodesSync(
+            final DistributedNativeConnectionProvider distributedNativeConnectionProvider
+    ) throws UnknownHostException, EcChronosException
+    {
+        LOG.info("Creating ecChronos nodes_sync bean");
+        EccNodesSync myEccNodesSync = EccNodesSync.newBuilder()
+                .withInitialNodesList(distributedNativeConnectionProvider.getNodes())
+                .withSession(distributedNativeConnectionProvider.getCqlSession())
+                .withEcchronosID(ecChronosID)
+                .build();
+        LOG.info("ecChronos nodes_sync bean created with success");
+        LOG.info("Starting to acquire nodes");
+        myEccNodesSync.acquireNodes();
+        LOG.info("Nodes acquired with success");
```

Comment:
Perhaps not so many infos is necessary? It will throw if it fails, so perhaps the last info is enough?

Reply:
ok